components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1421,4 +1427,164 @@ export class GitpodServerEEImpl extends GitpodServerImpl<GitpodClient, GitpodSer
         await this.subscriptionService.addCredit(user.id, 50, now);
         return 'Thank you for you feedback. We have added 50 Gitpod Hours to your account. Have fun!';
     }
+
+    //#region Projects concerns
+    //
+    async getProviderRepositoriesForUser(params: { provider: string, hints?: object }): Promise<ProviderRepository[]> {
+        const user = this.checkAndBlockUser("getProviderRepositoriesForUser");
```

Comment:
this method should go the respective GitProvider impl. We should not add dependencies to GitHub (octokit) across our code.

Reply:
+1 for moving out. I'm thinking of a `GitHubAppSupport` injectable.  

FYI we have two octokit dependencies
- first in our GitHub support (auth provider, context parsers, etc.)
- second comes in via probot
Both parts live in different contexts. More specifically: the GitHub support is instantiated per Git Provider host. 

BTW it's ~impossible~ impractical to sync both octokit versions both. Also, obtaining GH App's authenticated API (octokit) is done via `GitHubApp.server.probotApp.auth()`.